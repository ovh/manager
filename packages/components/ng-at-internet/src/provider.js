import OvhAtInternet from '@ovh-ux/ovh-at-internet';

/**
 * @ngdoc service
 * @require ATInternet smarttag.js library
 * @name atInternet
 * @description
 * Angular service wrapping the AtInternet tracking Javascript library.
 *
 * More informations : http://developers.atinternet-solutions.com/javascript-fr/bien-commencer-fr-javascript-fr/initialisation-du-tracker-javascript/
 */
/**
 * @ngdoc service
 * @require ATInternet smarttag.js library
 * @name atInternetProvider
 * @description
 * Provider allowing configuration for atInternet service.
 */
export default class NgAtInternet extends OvhAtInternet {
  $get() {
    return this;
  }
}
