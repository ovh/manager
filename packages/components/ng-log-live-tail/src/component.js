import template from './template.html';
import controller from './controller';

export default {
  controller,
  template,
  transclude: {
    tile: '?tile',
    description: '?description',
  },
  bindings: {
    logApiUrl: '<',
    logKind: '<',
    logKeys: '<',
  },
};
