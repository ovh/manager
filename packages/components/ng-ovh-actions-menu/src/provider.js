/**
 * @ngdoc service
 * @name actionsMenu.actionsMenuProvider
 *
 * @description
 * actionsMenuProvider allows developper to configure the path of the translation file.
 *
 * @example
 * <pre>
 *     angular.module("myManagerApp").config(function (actionsMenuProvider) {
 *          // set a new path for translations of the module
 *          actionsMenuProvider.setTranslationPath("/the/new/translation/path");
 *      });
 * </pre>
 */
export default function() {
  const self = this;
  let translationPath =
    '../bower_components/ovh-angular-actions-menu/dist/ovh-angular-actions-menu';

  /*= ====================================
    =            CONFIGURATION            =
    ===================================== */

  /**
   *  @ngdoc method
   *  @name actionsMenu.actionsMenuProvider#setTranslationPath
   *  @methodOf actionsMenu.actionsMenuProvider
   *
   *  @description
   *  Allows you to change the default location of the translation file of the module.
   *
   *  @param {String} path The new path of the translation file.
   *
   *  @return {String} The new configured translation path.
   */
  self.setTranslationPath = function(path) {
    if (path) {
      translationPath = path;
    }

    return translationPath;
  };

  /* -----  End of CONFIGURATION  ------*/

  /**
   *  @ngdoc service
   *  @name actionsMenu.service:actionsMenu
   *
   *  @description
   *  This service enable you to load translation file.
   */
  self.$get = /* @ngInject */ function(
    $translate /* , $translatePartialLoader */,
  ) {
    return {
      /**
       *  @ngdoc method
       *  @name actionsMenu.service:actionsMenu#
       *  @methodOf actionsMenu.service:actionsMenu
       *
       *  @description
       *  Load the translation file from the configured path.
       *
       *  @return {Promise} When the translation file is loaded.
       */
      loadTranslations() {
        // $translatePartialLoader.addPart(translationPath);
        return $translate.refresh();
      },
    };
  };
}
