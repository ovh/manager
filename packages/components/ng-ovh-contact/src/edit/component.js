import controller from './controller';
import template from './template.html';

export default {
  require: {
    ovhContactCtrl: '^ovhContact',
  },
  template,
  controller,
  bindings: {
    options: '=?ovhContactChoiceOptions',
    customList: '=?ovhContactChoiceCustomList',
  },
};
