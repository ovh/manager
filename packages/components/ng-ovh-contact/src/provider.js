import angular from 'angular';
import { forEach, has, keys, set, snakeCase } from 'lodash-es';

export default function() {
  const self = this;

  self.$get = /* @ngInject */ function $get(
    $q,
    $translate,
    OvhContact,
    OvhApiMe,
    CONTACT_EMAIL_REGEX,
    iceberg,
  ) {
    /**
     *  @ngdoc service
     *  @name ovhContact.service:ovhContact
     *
     *  @requires $q
     *  @requires $translate
     *  @requires OvhContact
     *  @requires OvhApiMe
     *
     *  @description
     *  The `ovhContact` service is the actual core of ovhContact module.
     *  This service manage the contacts of the connected user.
     */

    const ovhContactService = {};
    let contacts = [];
    let schemas = null;

    /**
     *  @private
     *  Get /me ovh api schema. Call GET /me.json.
     */
    function getApiSchemas() {
      if (!schemas) {
        return OvhApiMe.v6()
          .schema()
          .$promise.then((apiSchemas) => {
            schemas = apiSchemas;
            return schemas;
          });
      }
      return $q.when(schemas);
    }

    /**
     *  @private
     *  Convert nic address inforamtions to set contact.address informations.
     */
    function convertNicAddressToContactAddress(nic) {
      // keys are the contact address fields names / values are nic fields names
      const addressMapping = {
        line1: 'address',
      };
      const contactAddress = {};

      forEach(schemas.models['contact.Address'].properties, (value, key) => {
        if (has(addressMapping, key) && has(nic, addressMapping[key])) {
          contactAddress[key] = nic[addressMapping[key]];
        } else if (has(nic, key)) {
          contactAddress[key] = nic[key];
        } else {
          contactAddress[key] = null;
        }
      });

      return contactAddress;
    }

    /**
     *  @private
     *  Convert a nic to contact.
     */
    function convertNicToContact(nic) {
      // keys are the contact fields names / values are nic fields names
      const fieldMapping = {
        organisationName: 'organisation',
        lastName: 'name',
        gender: 'sex',
      };
      const contact = {};

      // iterate into contact model to set nic infos
      angular.forEach(
        keys(schemas.models['contact.Contact'].properties),
        (key) => {
          if (key === 'address') {
            contact.address = convertNicAddressToContactAddress(nic);
          } else if (key === 'birthDay') {
            if (new RegExp(/^(\d{1,2})\/(\d{1,2})\/(\d{4})$/).test(nic[key])) {
              const splittedDate = nic[key].split('/');
              contact[key] = splittedDate.reverse().join('-');
            } else {
              contact[key] = null;
            }
          } else if (
            fieldMapping[key] &&
            (has(nic, fieldMapping[key]) ||
              has(nic, fieldMapping[key.toLowerCase()]))
          ) {
            contact[key] =
              nic[fieldMapping[key]] ||
              nic[fieldMapping[key.toLowerCase()]] ||
              null;
          } else if (has(nic, key) || has(nic, key.toLowerCase())) {
            contact[key] = nic[key] || nic[key.toLowerCase()] || null;
          } else {
            contact[key] = null;
          }
        },
      );

      return contact;
    }

    /* ----------  CONTACT LIST MANAGEMENT  ----------*/

    /**
     *  @ngdoc method
     *  @name ovhContact.service:ovhContact#addContact
     *  @methodOf ovhContact.service:ovhContact
     *
     *  @description
     *  Add a contact into contacts list.
     *
     *  @param {Object|OvhContact} contactOptions An object with options
     *  or an instance of OvhContact.
     *
     *  @return {OvhContact} The added contact.
     */
    ovhContactService.addContact = function addContact(contactOptions) {
      const addedContact =
        contactOptions instanceof OvhContact
          ? contactOptions
          : new OvhContact(contactOptions);
      contacts.push(addedContact);
      return addedContact;
    };

    /**
     *  @ngdoc method
     *  @name ovhContact.service:ovhContact#getContactList
     *  @methodOf ovhContact.service:ovhContact
     *
     *  @description
     *  Get the entire list of contacts. This use api v7 to get all contacts details.
     *
     *  @param {Boolean} refresh Flag telling to fully recreate the contact list.
     *
     *  @return {Promise} That return an Array of OvhContact.
     */
    ovhContactService.getContactList = function getContactList(refresh) {
      const selff = this;

      if (refresh) {
        contacts = [];
      }

      return iceberg('/me/contact')
        .query()
        .expand('CachedObjectList-Pages')
        .execute()
        .$promise.then(({ data: contactsList }) => {
          // filter contact that are not already added
          // this avoid loosing contact object reference
          // then add contact to contact list (at given index)

          const contactsToAdd = contactsList.filter(
            (contact) => !contacts.some(({ id }) => contact.id === id),
          );

          forEach(contactsToAdd, (contact) => {
            selff.addContact(contact);
          });

          return contacts;
        });
    };

    /**
     *  @ngdoc method
     *  @name ovhContact.service:ovhContact#getCreationRules
     *  @methodOf ovhContact.service:ovhContact
     *
     *  @description
     *  Get the rules for creating a new ovh contact.
     *
     *  @return {Object} Representing the rules for creating an new contact.
     */
    ovhContactService.getCreationRules = function getCreationRules() {
      let models = null;

      function updateCreationRules(rules) {
        angular.forEach(rules, (value, key) => {
          if (angular.isDefined(models[value.fullType])) {
            if (models[value.fullType]) {
              if (angular.isArray(models[value.fullType].enum)) {
                const enumId = models[value.fullType].id
                  .replace(/Enum$/, '')
                  .toLowerCase();
                set(rules, `${key}.values`, []);
                angular.forEach(models[value.fullType].enum, (enumValue) => {
                  rules[key].values.push({
                    label: $translate.instant(
                      [
                        ['country', 'language'].indexOf(enumId) > -1
                          ? enumId
                          : snakeCase(`ovh_contact_edit_${enumId}`),
                        enumValue,
                      ].join('_'),
                    ),
                    value: enumValue,
                  });
                });
              } else if (angular.isObject(models[value.fullType].properties)) {
                angular.forEach(
                  updateCreationRules(models[value.fullType].properties),
                  (subValue, subKey) => {
                    // eslint-disable-next-line no-param-reassign
                    rules[`${key}.${subKey}`] = subValue;
                  },
                );
              }
            }
          }

          // TODO API hack, phone is mandatory
          if (key === 'phone') {
            set(value, 'canBeNull', 0);
          } else if (key === 'email') {
            set(value, 'regularExpression', CONTACT_EMAIL_REGEX);
          }
        });
        return rules;
      }

      return getApiSchemas().then(({ models: schemaModels }) => {
        models = schemaModels;
        return updateCreationRules(models['contact.Contact'].properties);
      });
    };

    /* ----------  CONNECTED USER  ----------*/

    /**
     *  @ngdoc method
     *  @name ovhContact.service:ovhContact#getConnectedUser
     *  @methodOf ovhContact.service:ovhContact
     *
     *  @description
     *  Get the connected user informations. Call GET /me API.
     *
     *  @return {Object} Representing the connected user.
     */
    ovhContactService.getConnectedUser = function getConnectedUser() {
      return OvhApiMe.v6().get().$promise;
    };

    /**
     *  @ngdoc method
     *  @name ovhContact.service:ovhContact#convertConnectedUserToContact
     *  @methodOf ovhContact.service:ovhContact
     *
     *  @description
     *  Convert the connected user nic to an ovh contact.
     *
     *  @return {Promise} That returns an OvhContact instance.
     */
    ovhContactService.convertConnectedUserToContact = function convertConnectedUserToContact() {
      return this.getConnectedUser().then((nic) =>
        getApiSchemas().then(() => new OvhContact(convertNicToContact(nic))),
      );
    };

    return ovhContactService;
  };
}
