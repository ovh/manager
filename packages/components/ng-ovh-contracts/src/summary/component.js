import template from './template.html';

const component = {
  bindings: {
    contracts: '<items',
    inlineContracts: '<inlineItems',
    model: '=',
    name: '@?',
  },
  template,
};

export default component;
