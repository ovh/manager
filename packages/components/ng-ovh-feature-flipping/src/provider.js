import FeatureFlipping from './service';

export default class NgOvhFeatureFlippingProvider {
  constructor() {
    this.applicationName = null;
  }

  /**
   * Set the application name that will request the 2API feature availability.
   *
   * @param {string} applicationName The current application name
   */
  setApplicationName(applicationName) {
    this.applicationName = applicationName;
  }

  /* @ngInject */
  $get($q, $http) {
    return new FeatureFlipping($q, $http, this.applicationName);
  }
}
