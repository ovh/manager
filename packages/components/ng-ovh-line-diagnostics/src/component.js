import controller from './controller';
import template from './template.html';

export default {
  bindings: {
    lineNumber: '@',
    serviceName: '@',
  },
  template,
  controller,
};
