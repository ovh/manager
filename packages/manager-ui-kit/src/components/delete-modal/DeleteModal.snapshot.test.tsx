import { vitest } from 'vitest';
import { render } from '../../utils/test.provider';
import { DeleteModal } from './DeleteModal.component';
import { DeleteModalProps } from './DeleteModal.props';

export const sharedProps: DeleteModalProps = {
  onClose: vitest.fn(),
  onConfirmDelete: vitest.fn(),
  serviceTypeName: 'serviceType',
  open: true,
};

describe('Delete Modal component', () => {
  it('renders basic modal', () => {
    const { container } = render(<DeleteModal {...sharedProps} />);
    expect(container.parentElement).toMatchSnapshot();
  });

  it('renders loading modal', () => {
    const { container } = render(<DeleteModal {...sharedProps} isLoading />);
    expect(container.parentElement).toMatchSnapshot();
  });

  it('renders modal with error', () => {
    const { container } = render(
      <DeleteModal {...sharedProps} error="Test error message" />,
    );
    expect(container.parentElement).toMatchSnapshot();
  });

  it('renders modal with custom service type', () => {
    const { container } = render(
      <DeleteModal {...sharedProps} serviceTypeName="Custom Service" />,
    );
    expect(container.parentElement).toMatchSnapshot();
  });

  it('renders modal with children content', () => {
    const { container } = render(
      <DeleteModal {...sharedProps}>
        <div>Additional content</div>
      </DeleteModal>,
    );
    expect(container.parentElement).toMatchSnapshot();
  });
});
