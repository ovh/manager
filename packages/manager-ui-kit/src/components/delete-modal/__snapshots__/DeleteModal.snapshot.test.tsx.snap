// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Delete Modal component > renders basic modal 1`] = `
<body>
  <div />
  <div
    class="_modal-backdrop_inusz_22"
    data-part="backdrop"
    data-scope="dialog"
    data-state="open"
    dir="ltr"
    id="dialog::r0::backdrop"
  />
  <div
    class="_modal-positioner_inusz_29"
    data-part="positioner"
    data-scope="dialog"
    dir="ltr"
    id="dialog::r0::positioner"
  >
    <div
      aria-describedby=""
      aria-labelledby=""
      aria-modal="true"
      class="_modal-content_inusz_35"
      data-ods="modal-content"
      data-part="content"
      data-scope="dialog"
      data-state="open"
      dir="ltr"
      id="dialog::r0::content"
      role="alertdialog"
      tabindex="-1"
    >
      <div
        class="_modal-header_1qhu0_2 _modal-header--critical_1qhu0_14"
      >
        <button
          aria-label="Close"
          class="_button_1koyw_2 _button--neutral_1koyw_83 _button--xs_1koyw_25 _button--ghost_1koyw_151 _modal-header__close_1qhu0_26"
          data-ods="button"
          data-part="close-trigger"
          data-scope="dialog"
          dir="ltr"
          id="dialog::r0::close"
          type="button"
        >
          <span
            class="_icon_10c23_2 _icon--xmark_10c23_562"
            data-ods="icon"
            role="presentation"
          />
        </button>
      </div>
      <div
        class="_modal-body_3om6u_2 text-left"
        data-ods="modal-body"
      >
        <h4
          class="_text--heading-4_1aw48_31 mb-4"
          data-ods="text"
        >
          Résilier serviceType ?
        </h4>
        <div
          class="flex flex-col text-left"
        >
          <div
            class="flex flex-col gap-4"
          >
            <p
              class="_text--paragraph_1aw48_54"
              data-ods="text"
              data-testid="manager-delete-modal-description"
              id="delete-modal-description"
            >
              La résiliation de votre service effacera définitivement toutes les données correspondantes. Souhaitez-vous procéder à la résiliation de votre service ?
            </p>
          </div>
        </div>
        <div
          class="flex justify-end flex-wrap gap-4"
        >
          <button
            class="_button_1koyw_2 _button--critical_1koyw_49 _button--md_1koyw_42 _button--ghost_1koyw_151 mt-4"
            data-ods="button"
            data-testid="manager-delete-modal-cancel"
            type="button"
          >
            Non, refuser
          </button>
          <button
            class="_button_1koyw_2 _button--critical_1koyw_49 _button--md_1koyw_42 _button--default_1koyw_49 mt-4"
            data-ods="button"
            data-testid="manager-delete-modal-confirm"
            type="button"
          >
            Oui, résilier
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Delete Modal component > renders loading modal 1`] = `
<body>
  <div />
  <div
    class="_modal-backdrop_inusz_22"
    data-part="backdrop"
    data-scope="dialog"
    data-state="open"
    dir="ltr"
    id="dialog::r1::backdrop"
  />
  <div
    class="_modal-positioner_inusz_29"
    data-part="positioner"
    data-scope="dialog"
    dir="ltr"
    id="dialog::r1::positioner"
  >
    <div
      aria-describedby=""
      aria-labelledby=""
      aria-modal="true"
      class="_modal-content_inusz_35"
      data-ods="modal-content"
      data-part="content"
      data-scope="dialog"
      data-state="open"
      dir="ltr"
      id="dialog::r1::content"
      role="alertdialog"
      tabindex="-1"
    >
      <div
        class="_modal-header_1qhu0_2 _modal-header--critical_1qhu0_14"
      >
        <button
          aria-label="Close"
          class="_button_1koyw_2 _button--neutral_1koyw_83 _button--xs_1koyw_25 _button--ghost_1koyw_151 _modal-header__close_1qhu0_26"
          data-ods="button"
          data-part="close-trigger"
          data-scope="dialog"
          dir="ltr"
          id="dialog::r1::close"
          type="button"
        >
          <span
            class="_icon_10c23_2 _icon--xmark_10c23_562"
            data-ods="icon"
            role="presentation"
          />
        </button>
      </div>
      <div
        class="_modal-body_3om6u_2 text-left"
        data-ods="modal-body"
      >
        <h4
          class="_text--heading-4_1aw48_31 mb-4"
          data-ods="text"
        >
          Résilier serviceType ?
        </h4>
        <div
          class="flex justify-center my-5"
          data-testid="spinner"
        >
          <span
            class="_spinner_o7c7y_10 _spinner--primary_o7c7y_29 _spinner--md_o7c7y_20"
            data-ods="spinner"
            role="progressbar"
          >
            <svg
              aria-hidden="true"
              class="_spinner__svg_o7c7y_32"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="24"
                width="24"
              />
              <g
                transform="translate(-237.694 -93.932)"
              >
                <path
                  d="M-5236.305-4108.069h0v-2a9.011,9.011,0,0,0,9-9,9.01,9.01,0,0,0-9-9,9.01,9.01,0,0,0-9,9h-2a10.928,10.928,0,0,1,3.222-7.779,10.928,10.928,0,0,1,7.778-3.221,10.929,10.929,0,0,1,7.779,3.221,10.932,10.932,0,0,1,3.221,7.777,10.928,10.928,0,0,1-3.221,7.778A10.931,10.931,0,0,1-5236.305-4108.069Z"
                  transform="translate(5486 4225)"
                />
                <circle
                  cx="1"
                  cy="1"
                  r="1"
                  transform="translate(248.694 114.932)"
                />
                <circle
                  cx="1"
                  cy="1"
                  r="1"
                  transform="translate(238.694 104.932)"
                />
              </g>
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Delete Modal component > renders modal with children content 1`] = `
<body>
  <div />
  <div
    class="_modal-backdrop_inusz_22"
    data-part="backdrop"
    data-scope="dialog"
    data-state="open"
    dir="ltr"
    id="dialog::r4::backdrop"
  />
  <div
    class="_modal-positioner_inusz_29"
    data-part="positioner"
    data-scope="dialog"
    dir="ltr"
    id="dialog::r4::positioner"
  >
    <div
      aria-describedby=""
      aria-labelledby=""
      aria-modal="true"
      class="_modal-content_inusz_35"
      data-ods="modal-content"
      data-part="content"
      data-scope="dialog"
      data-state="open"
      dir="ltr"
      id="dialog::r4::content"
      role="alertdialog"
      tabindex="-1"
    >
      <div
        class="_modal-header_1qhu0_2 _modal-header--critical_1qhu0_14"
      >
        <button
          aria-label="Close"
          class="_button_1koyw_2 _button--neutral_1koyw_83 _button--xs_1koyw_25 _button--ghost_1koyw_151 _modal-header__close_1qhu0_26"
          data-ods="button"
          data-part="close-trigger"
          data-scope="dialog"
          dir="ltr"
          id="dialog::r4::close"
          type="button"
        >
          <span
            class="_icon_10c23_2 _icon--xmark_10c23_562"
            data-ods="icon"
            role="presentation"
          />
        </button>
      </div>
      <div
        class="_modal-body_3om6u_2 text-left"
        data-ods="modal-body"
      >
        <h4
          class="_text--heading-4_1aw48_31 mb-4"
          data-ods="text"
        >
          Résilier serviceType ?
        </h4>
        <div
          class="flex flex-col text-left"
        >
          <div
            class="flex flex-col gap-4"
          >
            <p
              class="_text--paragraph_1aw48_54"
              data-ods="text"
              data-testid="manager-delete-modal-description"
              id="delete-modal-description"
            >
              La résiliation de votre service effacera définitivement toutes les données correspondantes. Souhaitez-vous procéder à la résiliation de votre service ?
            </p>
            <div>
              Additional content
            </div>
          </div>
        </div>
        <div
          class="flex justify-end flex-wrap gap-4"
        >
          <button
            class="_button_1koyw_2 _button--critical_1koyw_49 _button--md_1koyw_42 _button--ghost_1koyw_151 mt-4"
            data-ods="button"
            data-testid="manager-delete-modal-cancel"
            type="button"
          >
            Non, refuser
          </button>
          <button
            class="_button_1koyw_2 _button--critical_1koyw_49 _button--md_1koyw_42 _button--default_1koyw_49 mt-4"
            data-ods="button"
            data-testid="manager-delete-modal-confirm"
            type="button"
          >
            Oui, résilier
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Delete Modal component > renders modal with custom service type 1`] = `
<body>
  <div />
  <div
    class="_modal-backdrop_inusz_22"
    data-part="backdrop"
    data-scope="dialog"
    data-state="open"
    dir="ltr"
    id="dialog::r3::backdrop"
  />
  <div
    class="_modal-positioner_inusz_29"
    data-part="positioner"
    data-scope="dialog"
    dir="ltr"
    id="dialog::r3::positioner"
  >
    <div
      aria-describedby=""
      aria-labelledby=""
      aria-modal="true"
      class="_modal-content_inusz_35"
      data-ods="modal-content"
      data-part="content"
      data-scope="dialog"
      data-state="open"
      dir="ltr"
      id="dialog::r3::content"
      role="alertdialog"
      tabindex="-1"
    >
      <div
        class="_modal-header_1qhu0_2 _modal-header--critical_1qhu0_14"
      >
        <button
          aria-label="Close"
          class="_button_1koyw_2 _button--neutral_1koyw_83 _button--xs_1koyw_25 _button--ghost_1koyw_151 _modal-header__close_1qhu0_26"
          data-ods="button"
          data-part="close-trigger"
          data-scope="dialog"
          dir="ltr"
          id="dialog::r3::close"
          type="button"
        >
          <span
            class="_icon_10c23_2 _icon--xmark_10c23_562"
            data-ods="icon"
            role="presentation"
          />
        </button>
      </div>
      <div
        class="_modal-body_3om6u_2 text-left"
        data-ods="modal-body"
      >
        <h4
          class="_text--heading-4_1aw48_31 mb-4"
          data-ods="text"
        >
          Résilier Custom Service ?
        </h4>
        <div
          class="flex flex-col text-left"
        >
          <div
            class="flex flex-col gap-4"
          >
            <p
              class="_text--paragraph_1aw48_54"
              data-ods="text"
              data-testid="manager-delete-modal-description"
              id="delete-modal-description"
            >
              La résiliation de votre service effacera définitivement toutes les données correspondantes. Souhaitez-vous procéder à la résiliation de votre service ?
            </p>
          </div>
        </div>
        <div
          class="flex justify-end flex-wrap gap-4"
        >
          <button
            class="_button_1koyw_2 _button--critical_1koyw_49 _button--md_1koyw_42 _button--ghost_1koyw_151 mt-4"
            data-ods="button"
            data-testid="manager-delete-modal-cancel"
            type="button"
          >
            Non, refuser
          </button>
          <button
            class="_button_1koyw_2 _button--critical_1koyw_49 _button--md_1koyw_42 _button--default_1koyw_49 mt-4"
            data-ods="button"
            data-testid="manager-delete-modal-confirm"
            type="button"
          >
            Oui, résilier
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Delete Modal component > renders modal with error 1`] = `
<body>
  <div />
  <div
    class="_modal-backdrop_inusz_22"
    data-part="backdrop"
    data-scope="dialog"
    data-state="open"
    dir="ltr"
    id="dialog::r2::backdrop"
  />
  <div
    class="_modal-positioner_inusz_29"
    data-part="positioner"
    data-scope="dialog"
    dir="ltr"
    id="dialog::r2::positioner"
  >
    <div
      aria-describedby=""
      aria-labelledby=""
      aria-modal="true"
      class="_modal-content_inusz_35"
      data-ods="modal-content"
      data-part="content"
      data-scope="dialog"
      data-state="open"
      dir="ltr"
      id="dialog::r2::content"
      role="alertdialog"
      tabindex="-1"
    >
      <div
        class="_modal-header_1qhu0_2 _modal-header--critical_1qhu0_14"
      >
        <button
          aria-label="Close"
          class="_button_1koyw_2 _button--neutral_1koyw_83 _button--xs_1koyw_25 _button--ghost_1koyw_151 _modal-header__close_1qhu0_26"
          data-ods="button"
          data-part="close-trigger"
          data-scope="dialog"
          dir="ltr"
          id="dialog::r2::close"
          type="button"
        >
          <span
            class="_icon_10c23_2 _icon--xmark_10c23_562"
            data-ods="icon"
            role="presentation"
          />
        </button>
      </div>
      <div
        class="_modal-body_3om6u_2 text-left"
        data-ods="modal-body"
      >
        <h4
          class="_text--heading-4_1aw48_31 mb-4"
          data-ods="text"
        >
          Résilier serviceType ?
        </h4>
        <div
          class="flex flex-col text-left"
        >
          <div
            class="flex flex-col gap-4"
          >
            <div
              class="_message_1oosv_2 _message--warning_1oosv_32 _message--default_1oosv_9"
              data-ods="message"
            >
              <span
                class="_icon_10c23_2 _icon--triangle-exclamation_10c23_538 _message-icon_mpgx8_2"
                data-ods="message-icon"
                role="presentation"
              />
              <div
                class="_message-body_fe7x3_2"
                data-ods="message-body"
              >
                L'erreur suivante est survenue : Test error message.
              </div>
              <button
                aria-label="Close"
                class="_button_1koyw_2 _button--neutral_1koyw_83 _button--xs_1koyw_25 _button--ghost_1koyw_151 _message__close_1oosv_25"
                data-ods="button"
                type="button"
              >
                <span
                  class="_icon_10c23_2 _icon--xmark_10c23_562"
                  data-ods="icon"
                  role="presentation"
                />
              </button>
            </div>
            <p
              class="_text--paragraph_1aw48_54"
              data-ods="text"
              data-testid="manager-delete-modal-description"
              id="delete-modal-description"
            >
              La résiliation de votre service effacera définitivement toutes les données correspondantes. Souhaitez-vous procéder à la résiliation de votre service ?
            </p>
          </div>
        </div>
        <div
          class="flex justify-end flex-wrap gap-4"
        >
          <button
            class="_button_1koyw_2 _button--critical_1koyw_49 _button--md_1koyw_42 _button--ghost_1koyw_151 mt-4"
            data-ods="button"
            data-testid="manager-delete-modal-cancel"
            type="button"
          >
            Non, refuser
          </button>
          <button
            class="_button_1koyw_2 _button--critical_1koyw_49 _button--md_1koyw_42 _button--default_1koyw_49 mt-4"
            data-ods="button"
            data-testid="manager-delete-modal-confirm"
            type="button"
          >
            Oui, résilier
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
